htdocs/core/triggers/interface_50_modEventOrganization_EventOrganization.class.php

```diff
@@ -0,0 +1,123 @@
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ *	\file       htdocs/core/triggers/interface_50_modEventOrganization_EventOrganization.class.php
+ *  \ingroup    eventorganization
+ *  \brief      Trigger file for Event Organization module
+ */
+
+require_once DOL_DOCUMENT_ROOT.'/core/triggers/dolibarrtriggers.class.php';
+
+
+/**
+ *  Class of triggered functions for agenda module
+ */
+class InterfaceEventOrganization extends DolibarrTriggers
+{
+	/**
+	 * Constructor
+	 *
+	 * @param DoliDB $db Database handler
+	 */
+	public function __construct($db)
+	{
+		$this->db = $db;
+
+		$this->name = preg_replace('/^Interface/i', '', get_class($this));
+		$this->family = "eventorganization";
+		$this->description = "Triggers of this module to manage event organization triggers action";
+		// 'development', 'experimental', 'dolibarr' or version
+		$this->version = self::VERSION_DOLIBARR;
+		$this->picto = 'action';
+	}
+
+	/**
+	 * Function called when a Dolibarrr business event is done.
+	 * All functions "runTrigger" are triggered if file is inside directory htdocs/core/triggers or htdocs/module/code/triggers (and declared)
+	 *
+	 * Following properties may be set before calling trigger. The may be completed by this trigger to be used for writing the event into database:
+	 *      $object->actiontypecode (translation action code: AC_OTH, ...)
+	 *      $object->actionmsg (note, long text)
+	 *      $object->actionmsg2 (label, short text)
+	 *      $object->sendtoid (id of contact or array of ids of contacts)
+	 *      $object->socid (id of thirdparty)
+	 *      $object->fk_project
+	 *      $object->fk_element	(ID of object to link action event to)
+	 *      $object->elementtype (->element of object to link action to)
+	 *      $object->module (if defined, elementtype in llx_actioncomm will be elementtype@module)
+	 *
+	 * @param string		$action		Event action code ('CONTRACT_MODIFY', 'RECRUITMENTCANDIDATURE_MODIFIY', ...)
+	 * @param Object		$object     Object
+	 * @param User		    $user       Object user
+	 * @param Translate 	$langs      Object langs
+	 * @param conf		    $conf       Object conf
+	 * @return int         				<0 if KO, 0 if no triggered ran, >0 if OK
+	 */
+	public function runTrigger($action, $object, User $user, Translate $langs, Conf $conf)
+	{
+		if (empty($conf->eventorganization->enabled)) {
+			return 0; // Module not active, we do nothing
+		}
+		$error=0;
+		// Actions
+		if ($action == 'PROJECT_VALIDATE') {
```

Comment:
This actions must be done only if project has the role "Use to organize event". Bu i see no test on the $object->use_...

Reply:
True, I've thix this and better transaction mangement